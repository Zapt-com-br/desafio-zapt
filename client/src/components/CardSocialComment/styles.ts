import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 38.4rem;
    break-inside: avoid-column;
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    border-radius: ${theme.border.radius};

    background: ${theme.colors.moonWhite};

    p {
      font-size: ${theme.font.sizes.medium};
      line-height: 137%;

      letter-spacing: -0.01em;

      color: ${theme.colors.arcadeGlow};

      ::after {
        content: '”';
      }

      ::before {
        content: '“';
      }
    }
  `}
`

export const UserInfo = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    gap: 12px;
    margin-top: ${theme.spacings.xsmall};

    img {
      width: 36px;
      height: 36px;
      object-fit: cover;
      border-radius: 50%;
      border: 1px solid ${theme.colors.primary};
    }

    span {
      font-weight: ${theme.font.medium};
      font-size: ${theme.font.sizes.small};
      line-height: 129%;

      letter-spacing: -0.01em;

      color: ${theme.colors.midnightDreams};
    }
  `}
`
