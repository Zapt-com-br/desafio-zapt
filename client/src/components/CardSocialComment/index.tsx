import Image from 'next/image'
import * as S from './styles'

type UserType = {
  name: string
  avatar_url: string
  username: string
  social_provider: string
}

type Testimonial = {
  text: string
}

export type CardSocialCommentProps = {
  user: UserType
  testimonial: Testimonial
}

const CardSocialComment = ({ testimonial, user }: CardSocialCommentProps) => (
  <S.Wrapper>
    <p>{testimonial.text}</p>
    <S.UserInfo title={`${user.name} via ${user.social_provider}`}>
      <Image
        src={user.avatar_url}
        alt={`Foto de perfil de ${user.name}`}
        width={36}
        height={36}
        objectFit="cover"
      />
      <span>@{user.username}</span>
    </S.UserInfo>
  </S.Wrapper>
)

export default CardSocialComment
