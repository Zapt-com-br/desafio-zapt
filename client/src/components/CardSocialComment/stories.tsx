import { Story, Meta } from '@storybook/react'
import CardSocialComment, { CardSocialCommentProps } from '.'

export default {
  title: 'CardSocialComment',
  component: CardSocialComment,
  args: {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incr√≠veis'
    }
  }
} as Meta

export const Default: Story<CardSocialCommentProps> = (args) => (
  <CardSocialComment {...args} />
)
