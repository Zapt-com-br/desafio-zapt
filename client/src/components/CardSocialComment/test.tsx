import { screen } from '@testing-library/react'
import renderWithTheme from 'utils/tests/test-util'

import CardSocialComment from '.'

const props = {
  user: {
    name: 'Alan Gabriel',
    avatar_url: '/img/profile.jpg',
    username: 'eualangabriel',
    social_provider: 'Twitter'
  },
  testimonial: {
    text:
      'Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incr√≠veis'
  }
}

describe('<CardSocialComment />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<CardSocialComment {...props} />)

    expect(screen.getByText(props.testimonial.text)).toBeInTheDocument()

    expect(
      screen.getByTitle(`${props.user.name} via ${props.user.social_provider}`)
    ).toBeInTheDocument()

    expect(screen.getByText(`@${props.user.username}`)).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: `Foto de perfil de ${props.user.name}` })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
