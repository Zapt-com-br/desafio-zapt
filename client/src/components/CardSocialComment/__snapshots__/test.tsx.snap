// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSocialComment /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 38.4rem;
  padding: 2.4rem 1.6rem;
  border-radius: 0.4rem;
  background: #EBF4FD;
}

.c0 p {
  font-size: 1.6rem;
  line-height: 137%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: #011FBB;
}

.c0 p::after {
  content: '”';
}

.c0 p::before {
  content: '“';
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
  margin-top: 1.6rem;
}

.c1 img {
  width: 36px;
  height: 36px;
  object-fit: cover;
  border-radius: 50%;
  border: 1px solid #334BC8;
}

.c1 span {
  font-weight: 500;
  font-size: 1.4rem;
  line-height: 129%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: #001C33;
}

<div
  class="c0"
>
  <p>
    Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incríveis
  </p>
  <div
    class="c1"
    title="Alan Gabriel via Twitter"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <noscript />
      <img
        alt="Foto de perfil de Alan Gabriel"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
    </div>
    <span>
      @
      eualangabriel
    </span>
  </div>
</div>
`
